Services/DataX.ServiceHost/DataX.ServiceHost.AspNetCore/Extensions/DataXServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,63 @@
+﻿using DataX.ServiceHost.AspNetCore.Startup;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Text;
+
+namespace DataX.ServiceHost.AspNetCore.Extensions
+{
+    /// <summary>
+    /// Extensions for IServiceCollection
+    /// </summary>
+    public static class DataXServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds the specified DataX startup filter to the provides services.
+        /// </summary>
+        /// <typeparam name="TStartup">A type deriving from <see cref="DataXServiceStartup"/></typeparam>
+        /// <param name="services">The services to add the DataX startup to</param>
+        /// <param name="startup">Optional startup instance</param>
+        /// <returns>The configured services</returns>
+        public static IServiceCollection AddDataXStartup<TStartup>(this IServiceCollection services, TStartup startup = null)
+            where TStartup : DataXServiceStartup, new()
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            startup = startup ?? new TStartup();
+
+            startup.ConfigureServices(services);
+
+            // Add the FlowManagement Startup to be called _along with_ using Configure in the WebHostBuilder
+            services.AddTransient<IStartupFilter>(_ => startup);
+
+            return services;
+        }
+
+        /// <summary>
+        /// Removes DataX controllers based on a provided DataX startup type
+        /// </summary>
+        /// <typeparam name="TStartup">Concrete startup type for assembly reference</typeparam>
+        /// <param name="manager">The application part manager to remove DataX controllers from</param>
+        /// <returns>The modified application part manager</returns>
+        public static ApplicationPartManager RemoveDataXController<TStartup>(this ApplicationPartManager manager)
```

Comment:
Good question. This is for flexible usage by a consumer.

Let's say I need to wrap my instance of DataX with some custom framework (indeed, this is the case) that has custom ILoggers, auth, etc. I have a couple of options for wrapping:
1. Create multiple projects referencing each of the DataX services with the same scaffolding
2. Reference all of the DataX services in a single project and have the runtime select its role via deployment configuration

In case 2, all DataX controllers for all services would be pulled in by MVC. Controllers can be removed via MVC's `ApplicationPartsManager`. This operation can be done manually by the consumer referencing the assemblies by hand, but I figured a convenience method would be better to have.

As for usage, it's not used within DataX itself.

@vijayupadya Does this help clarify? What are your thoughts?

Reply:
Yes, makes sense.